include/nihstro/shader_bytecode.h

```diff
@@ -663,6 +663,19 @@ union SwizzlePattern {
         w = 3
     };
 
+    /**
+     * Gets the raw 8-bit selector for the specified (1-indexed) source register.
+     */
+    unsigned GetRawSelector(unsigned src) const {
+        if (src < 0 || src > 3)
```

Comment:
the "src < 0" check is pointless since it'll always be false because src is unsigned... Also, you meant to write `src == 0 || src > 3`, didn't you?